refactor(home component): separate css from js code and put them into different files

Mloweedgar opened the pull request:
## Description
Extract css for Home component and place them in a dedicated file, add tests file for writing unit tests. all files associated with Home component(js,styles.js,test.js files) have been placed under Home folder.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** 
 - [x] Refactoring (no functional changes, no api changes)
 - [x] Code style update (formatting, local variables)



## Issue

**What is the current behavior?**
All source code for rendering Home component was placed in one file `Home.js`

**What is the new behavior?**

source code for rendering Home component  have been refactored into three separate files and all the files have been placed under `Home` folder

1. `Home.js` - react code for home component
2. `Home.styles.js` - styles for the home component
3. `Home.test.js` - code for testing home component


## Breaking change
**Does this PR introduce a breaking change?** 
 - [x] No

dadiorchen commented on src/components/Home/styles.css:
```diff
@@ -0,0 +1,33 @@
+.Home {
+  width: 100%;
+  height: 100%;
+  position: relative;
+}
```
@Mloweedgar If we use 'Home' as the class name, is it possible that we defined this the same name somewhere, so it would lead to conflict/override?
Like I define 'Home' as the class name in another component?

Mloweedgar replied:
Yes, there is a possibility @dadiorchen , My Assumption is that names for our react components are unique (eg. the will be only one Home component), I therefore use the component name as a namespace for my css classes such that all css classes for styling home component will be preceded by 'Home' to make sure those styles only apply to Home component.

Mloweedgar replied:
My idea for improving this could be us decide to use a tool that will ensure every css class we define is made unique, we could for example decide to use tool like [CSS Modules](https://github.com/css-modules/css-modules) or any other similar tool

I will also be happy to hear your advice on this @dadiorchen

dadiorchen replied:
I don't have any idea for the time being, but I think sperate the code and style is good for a clear and re-useable codebase.

dadiorchen replied:
Good job @Mloweedgar

Mloweedgar replied:
Thanks @dadiorchen!

nmcharlton commented:
@thejoshcooter Would you mind taking at look at Edgar's refactor PR to check it reflects the code structure you have in mind?

nmcharlton commented on src/components/common/GreenStandSvgLogo.js:
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
```
I think that would be a good idea. We have an `images` directory that's currently under `components` but should probably be directly under `src`.

nmcharlton commented on src/components/Home/Home.styles.js:
```diff
@@ -0,0 +1,45 @@
+  welcomeBox: {
+    height: '100%',
+    //padding: theme.spacing(4),
+    //padding: '40px',
```
Yes

nmcharlton commented on src/components/Home/Home.js:
```diff
@@ -0,0 +1,80 @@
+  const appContext = useContext(AppContext);
+
+  useEffect(() => {
+    document.title = `${documentTitle}`; // update page title
```
That's an interesting suggestion. How would you distinguish _leftover_ comments from legitimate, useful comments?

thejoshcooter replied:
We could set up es lint to lint for comments that would prevent pushing with comments, and then when there is a need for a useful comment you can explicitly tell es lint to ignore the no comment rule for that line or file.

It would just help keep things clean by preventing unnecessary comments, left over comments to oneself, and code that is commented out while working that is no longer needed but people often forget to remove.

nmcharlton replied:
I like the sentiment, but I'm concerned that it would discourage (particularly new) contributors from commenting.
I'd rather have too many comments than not enough.

Clearing up commented out code is a fairly straightforward activity; we could make it part of the code review and encourage developers to remove unused, commented-out code as they go along as standard code maintenance practice.

thejoshcooter replied:
That makes sense to me - leaving it to code review and building habits that way works too!

Mloweedgar commented:
@thejoshcooter thank you for the code review and quick comments, I have addressed your comments, incase I have skipped any of your input, please let me  know, I will be more than happy to address it.
